Ginger/Ginger/BusinessFlowPages_New/AddActionMenu/POMNavPage.xaml.cs

```diff
@@ -41,30 +89,80 @@ public TreeView1 TreeView
         {
             InitializeComponent();
 
+            mContext = context;
+            mItemTypeRootNode = itemTypeRootNode;
             GingerHelpProvider.SetHelpString(this, itemTypeName.TrimEnd(new char[] { 's' }));
 
             xTreeView.TreeTitle = itemTypeName;
             xTreeView.TreeIcon = itemTypeIcon;
 
+            mContext.PropertyChanged -= MContext_PropertyChanged;
+            mContext.PropertyChanged += MContext_PropertyChanged;
+
+            xTreeView.Tree.TreeNodesFilterByField = new Tuple<string, string>(nameof(ApplicationPOMModel.TargetApplicationKey) + "." + nameof(ApplicationPOMModel.TargetApplicationKey.ItemName), mContext.BusinessFlow.CurrentActivity.TargetApplication);
+            xTreeView.Tree.FilterType = UCTreeView.eFilteroperationType.Equals;
             TreeViewItem r = xTreeView.Tree.AddItem(itemTypeRootNode);
+
             r.IsExpanded = true;
 
             itemTypeRootNode.SetTools(xTreeView);
             xTreeView.SetTopToolBarTools(saveAllHandler, addHandler);
 
             xTreeView.Tree.ItemSelected += MainTreeView_ItemSelected;
-
             SetElementsGridView();
+
             //if (treeItemDoubleClickHandler != null)
             //{
             //    xTreeView.Tree.ItemDoubleClick += treeItemDoubleClickHandler;
             //}
         }
 
+        private void MContext_PropertyChanged(object sender, System.ComponentModel.PropertyChangedEventArgs e)
+        {
+            mContext.Activity.PropertyChanged -= Activity_PropertyChanged;
+            mContext.Activity.PropertyChanged += Activity_PropertyChanged;
+            if (e.PropertyName is nameof(mContext.BusinessFlow) || e.PropertyName is nameof(mContext.Activity))
+            {
+                UpdatePOMTree();
+            }
+        }
+
+        private void Activity_PropertyChanged(object sender, System.ComponentModel.PropertyChangedEventArgs e)
+        {
+            if (e.PropertyName == nameof(Activity.TargetApplication))
+            {
+                UpdatePOMTree();
+            }
+        }
+
+        private void UpdatePOMTree()
+        {
+            if (mContext.BusinessFlow.CurrentActivity == null)
+                mContext.BusinessFlow.CurrentActivity = mContext.BusinessFlow.Activities[0];
+
+            xTreeView.Tree.TreeNodesFilterByField = new Tuple<string, string>(nameof(ApplicationPOMModel.TargetApplicationKey) + "." + nameof(ApplicationPOMModel.TargetApplicationKey.ItemName), mContext.BusinessFlow.CurrentActivity.TargetApplication);
+            xTreeView.Tree.FilterType = UCTreeView.eFilteroperationType.Equals;
+            xTreeView.Tree.RefresTreeNodeChildrens(mItemTypeRootNode);
+        }
+
+        private void CurrentActivity_PropertyChanged(object sender, System.ComponentModel.PropertyChangedEventArgs e)
```

Comment:
CurrentActivity_PropertyChanged? needed?

Reply:
Not needed, updated with ContextChanged on latest push.